### PR TITLE
feat(fastlane): add fastlane spec

### PR DESCRIPTION
**Pull request by hangai247**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
feature
**What is the current behavior? (You can also link to an open issue here)**

**What is the new behavior (if this is a feature change)?**
[Fastlane](https://fastlane.tools/)  Automation tool for beta deployments and releases for iOS and Android apps.
**Additional info:**

### CODE REVIEW

**Comment by hangai247** [comment]
> Hello @hangai247, thank you very much for creating a Pull Request! Here is a small checklist to get this PR merged as quickly as possible:
> 
> * [x ]  Do all subcommands / options which take arguments include the `args` property (`args: {}`)?
> * [ ]  Are all options modular? E.g. `-a` `-u` `-x` instead of `-aux`
> * [ ]  Have all other checks passed?
> 
> Please add a 👍 as a reaction to this comment to show that you read this.

File: src/fastlane.ts [hunk 1]
```diff
@@ -0,0 +1,1188 @@
+const actionGenerators: Fig.Generator = {
+  custom: async (context) => {
+    const searchTerm = context[context.length - 1];
+    return [
```
Comment (fedeci) [inline src/fastlane.ts:4]: Here we want each action to have the subcommands below, right?
```ts
  interface ScriptsRow {
    name: string
    deprecated: boolean,
    description: string
  }
  ...
  script: "fastlane actions",
  postProcess: async (out) => {
    // we need to perform some kind of parsing to get the rows and columns of the table.
   rows.map((r: ScriptsRow[]) => {
     return {
       name: r.name,
       ...( r.deprecated && { deprecated: r.deprecated }),
       description: r.description,
       subcommands: [] // The subcommands listed below
     }
   })
  }
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Thank you, I left a couple of comments, let me know if you need help!

File: src/fastlane.ts (hunk 1 above)
Comment (hangai247) [reply src/fastlane.ts:4]: @fedeci Thank you for your review. I will fix all of the above implicit configs. :)

About the action generator. Actually, I was trying to do the same. But unfortunately, fastlane actions and plugins command response is returning some weird markdown txt response, which I couldn't figure out easily.

Comment (fedeci) [reply src/fastlane.ts:4]: Yeah, that's a super weird output. We may try to propose the fastlane team to add a --json flag to the command to get a nicer output WDYT?

Comment (hangai247) [reply src/fastlane.ts:4]: Sounds very good. Thank you very much. I will remove this part for now and I hope we can incrementally improve it in the future. 😄

**Review by hangai247 — COMMENTED** [review]
Approved

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**